x-pack/plugins/observability_solution/observability_ai_assistant_app/server/functions/visualize_esql.ts

```diff
@@ -64,15 +64,20 @@ export function registerVisualizeESQLFunction({
           meta: { type: esFieldTypeToKibanaFieldType(type) },
         })) ?? [];
 
-      const message = getMessageForLLM(intention, query, Boolean(errorMessages.length));
+      // there is an actual error only when the query hasn't returned results
+      // and the client side validation has detected errors, we are adding the
+      // extra columns check as the client side validator can return false positives
+      const queryHasActualErrors = Boolean(errorMessages.length && !columns.length);
+
+      const message = getMessageForLLM(intention, query, queryHasActualErrors);
 
       return {
         data: {
           columns,
         },
         content: {
           message,
-          errorMessages,
+          errorMessages: queryHasActualErrors ? errorMessages : [],
```

Comment:
can we remove this key entirely if it has no errors? e.g. `...(queryHasActualErrors ? { errorMessages } : {})`. Otherwise the presence of the key might cause the LLM to hallucinate

Reply:
Natuurlijk. I didn't know that they are SO smart 😄 [Done](https://github.com/elastic/kibana/pull/181218/commits/7e91405e61a96352c9ad823b22bf1b0ff80bfbac)